### DISCUSSION TITLE
reticulum experiments

### DISCUSSION THREAD

**Discussion by milesif** [markqvist/Reticulum#571] [Help & Questions] [answered]
Hi everybody,

I am starting to experiment with reticulum. My final goal is to have a LoRa connecting to a friend home network, go through a server with a public address, reach my home network e finally to a second LoRa.

I set up 2 docker instances (D1 and D2) on the same ip network on my home pc. After enabling IP6 I had them talking to each other both with nomadnet and rnsh. So I expanded my experiments.

1.  I setup reticulum on a machine with a public ip (SRV1) configured as
                enable_transport = True
    
             [[Default Interface]]
                type = AutoInterface
                enabled = Yes

            [[TCP Server Interface]]
                type = TCPServerInterface
                interface_enabled = True
                listen_ip = 0.0.0.0
                listen_port = 23456
2.  I configured D1 as 

             [[Default Interface]]
                type = AutoInterface
                enabled = Yes

            [[TCP Client Interface]]
                type = TCPClientInterface
                interface_enabled = True
                target_host = <ip of public host>
                target_port = 23456

I could connect with rnsh from D1 to SRV1.

Then I set "enable_transport = Yes" on D1 and tried to connect from D2 to SRV1, assuming D2 would use D1 as transport node.
But after launching "rnsh -v <SRV1 destination>" I see "Requesting path..." forever and nothing happens.

Could anybody help me pointing out where I am wrong or what I should do to debug the problem?

thanks a lot in advance.

**Comment by faragher** [comment]
It should work. If D1 knows the path, it should respond to a request from D2. Have you restarted rnsd since you changed the configuration? 

When in doubt, you can try announcing with SRV1 again just to make sure at least D1 and hopefully D2 have seen the announce. 

Can you try using another app like NomadNet to test communication in general? Does rnstatus tell you the interfaces are up?

**Comment by milesif** [comment, marked as answer]
I tried with NomadNet and it worked. At first I could send messages from D1 to SRV1 and on SRV1 I got the following screen

![image](https://github.com/user-attachments/assets/94963c7c-e6f8-46c0-b1ac-be2af3a286ba)

After a while the keys were received by SRV1 and now the communication works in both directions...

I can go on with the rest of the setup. I will also try to make some project  examples work over these setup.

thanks a lot! especially for the super-fast answer

**Reply by faragher** [reply]
Glad to help! Let us know if you have any other questions.
